tests/python/test_python_frontend.py

```diff
@@ -4325,6 +4325,38 @@ def fusion_func(fd: FusionDefinition) -> None:
 
         self.exec_nvfuser(fusion_func, inputs)
 
+    def test_slice_api(self):
+        x = torch.randn((2, 5, 10), dtype=torch.float32, device="cuda:0")
+
+        offset = (0, 1, 2)
+
+        def fusion_func(fd: FusionDefinition) -> None:
+            T0 = fd.define_tensor(
+                shape=[-1, -1, -1],
+                contiguity=[True, True, True],
+                dtype=DataType.Float,
+                is_cpu=False,
+                stride_order=[2, 1, 0],
+            )
+            T1 = fd.ops.slice(
+                T0, start_indices=offset, end_indices=(2, 5, 10), strides=(1, 1, 1)
+            )
+            fd.add_output(T1)
+            V_start = fd.define_vector(offset)
+            V_end = T0.shape()
```

Comment:
Off-topic: would it be better to bind `Tensor.shape` using [def_property](https://pybind11.readthedocs.io/en/stable/classes.html#instance-and-static-fields) so we could use `T0.shape` instead of `T0.shape()`?

Reply:
I don't have a strong opinion one way or the other. :stuck_out_tongue_winking_eye:

Reply:
Me either, so let's leave it as is :-)